### PR TITLE
Fix Hardcoded Currency In System Tests

### PR DESCRIPTION
**Pull request by tubbo**
Refactor system tests to not look for a currency when possible, and when
not possible, read the default currency symbol from `Money` instead of
assuming it is `$`.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -9,5 +9,5 @@ gem 'workarea-testing', path: 'testing'
 gem 'listen'
 gem 'bundler-audit'
 gem 'rubocop'
-gem 'rails-decorators', git: 'https://github.com/workarea-commerce/rails-decorators.git'
+gem 'rails-decorators', git: 'https://github.com/workarea-commerce/rails-decorators.git', branch: 'master'
```
Comment (tubbo) [inline Gemfile]: I can remove this if you folks want me to, but this makes it a bit easier to work with master branch of workarea since I have a local override on my system for rails-decorators so it's a bit less cumbersome to work with experimental branches across multiple repos.

File: testing/lib/workarea/storefront/system_test.rb [hunk 2]
```diff
@@ -151,6 +151,10 @@ def find_analytics_events(for_event: nil)
           all_events.select { |e| e['name'] == for_event }
         end
       end
+
+      def currency
+        Money.default_currency.symbol
```
Comment (tubbo) [inline testing/lib/workarea/storefront/system_test.rb]: Wouldn't say I'm "in love" with this approach because I'm not sure how it will jive with multi-site apps that use different currencies per site...

**Comment by tubbo** [comment]
there's one remaining admin test failure: https://github.com/workarea-commerce/workarea/pull/349/checks?check_run_id=424164062

```
Failure:
Workarea::Admin::PublishAuthorizationSystemTest#test_user_cannot_select_publish_now_in_workflows [/Users/tscott/Code/workarea/admin/test/system/workarea/admin/publish_authorization_system_test.rb:54]:
expected to find text "Success" in "Shortcuts\n4 Alerts\nHelp\nwww.example.com\nEnglish\nTest Category: Add Content\nor skip this\nView on Storefront\nTest Category\nAbove Results\nBelow Results\nAdvanced\n← Cancel\nHTML\nSettings\nDisplay\nHTML\n<h3 style=\"font-size: 40px;\">Raw HTML Content</h3>\nPublishing now\nPublish with Publishing Test\na new release\n→\nSave\nCancel\n1) Setup 2) Products 3) Content 4) Taxonomy 5) Publish\nContinue to Taxonomy →"
```

(changed it to `assert_text` so it's actually possible to view)

The screenshot shows there's definitely no flash message. Is this something only I am getting or is anyone else getting this error? I figured master branch might be a bit unstable these days due to the rails 6 thing...

File: storefront/test/system/workarea/storefront/categories_system_test.rb [hunk 3]
```diff
@@ -175,7 +177,8 @@ def test_filtering_products_and_sorting
         visit storefront.category_path(category)
 
         Capybara.match = :first
-        click_link '$10.00 - $19.99 (1)'
+        price_range = "#{currency}10.00 - #{currency}19.99"
```
Comment (mttdffy) [inline storefront/test/system/workarea/storefront/categories_system_test.rb]: not all currencies have a prefix symbol, no?

Comment (mttdffy) [reply storefront/test/system/workarea/storefront/categories_system_test.rb]: maybe I'm wrong, but in any case, you're doing this in a lot of places, I'd recommend creating a factory method for this.

File: Gemfile (hunk 1 above)
Comment (bencrouse) [reply Gemfile]: I don't quite follow - aren't these two lines equivalent given `rails-decorators` default branch is `master`?

File: storefront/test/system/workarea/storefront/categories_system_test.rb (hunk 3 above)
Comment (mttdffy) [reply storefront/test/system/workarea/storefront/categories_system_test.rb]: might be best to find a way to use `number_to_currency` so that behavior of generating this string matches what happens on the storefront

File: admin/test/system/workarea/admin/publish_authorization_system_test.rb [hunk 4]
```diff
@@ -51,7 +51,7 @@ def test_user_cannot_select_publish_now_in_workflows
         click_link 'HTML'
         click_button 'create_block'
 
-        assert(page.has_content?('Success'))
+        assert_text('Success')
```
Comment (bencrouse) [inline admin/test/system/workarea/admin/publish_authorization_system_test.rb]: This is a bit of an unrelated change.

File: testing/lib/workarea/storefront/system_test.rb (hunk 2 above)
Comment (bencrouse) [reply testing/lib/workarea/storefront/system_test.rb]: Yeah... I'm not loving that either. Did you try or consider refactoring to make the actual symbol irrelevant? Like just test for the amounts. That seems like it would be cleaner in the end. Most if not all these assertions could changed to do that.

More complex ones could be regex like

```
assert(page.has_content?("Next Day - #{currency}20.00"))
# converts to
assert_match(page.text, /Next Day - .20\.00/)
```

or something like that.

File: Gemfile (hunk 1 above)
Comment (tubbo) [reply Gemfile]: @bencrouse sorry, never actually mentioned that I was receiving an error when you run `bundle` if you have a local override set up for the gem and it's pointing to a Git repo. Basically, since the Bundle is depending on `master` branch, the local override needs to also have `master` branch checked out in order to make sure the code is the same.

File: storefront/test/system/workarea/storefront/categories_system_test.rb (hunk 3 above)
Comment (tubbo) [reply storefront/test/system/workarea/storefront/categories_system_test.rb]: I didn't think `number_to_currency` was available in tests...

File: admin/test/system/workarea/admin/publish_authorization_system_test.rb (hunk 4 above)
Comment (tubbo) [reply admin/test/system/workarea/admin/publish_authorization_system_test.rb]: this was for debugging, the test here failed and i'm still not 100% on why.

File: testing/lib/workarea/storefront/system_test.rb (hunk 2 above)
Comment (tubbo) [reply testing/lib/workarea/storefront/system_test.rb]: TIL, `assert_text` takes a regex! :) https://rubydoc.info/github/jnicklas/capybara/master/Capybara/Node/Matchers#has_text%3F-instance_method

**Comment by tubbo** [comment]
I can't figure out how to `click_link` on some of the price filters. I tried the following:

```ruby
click_link '.result-filters__section--price .result-filters__group result-filters__link'
```

and

```ruby
within '.result-filters__section--price .result-filters__group' do
  click_link '.result-filters__link'
end
```

While the `within` call confirmed that the containing elements were being found, selecting the actual link doesn't seem to work.
